docs/guiding-philosophy.adoc

```diff
@@ -0,0 +1,21 @@
+= Cypher Design Guiding Philosophy
+
+To guarantee the consistent development of Cypher as a language, with many actors and designers involved, a sort of manifesto for Cypher was developed.
```

Comment:
I am not overly fond of "a sort of manifesto" (a bit hand-wavey and clumsy)... something such as "an informal manifesto" sounds better